web/src/beta/lib/reearth-ui/components/TextInput/index.tsx

```diff
@@ -61,40 +73,62 @@ export const TextInput: FC<TextInputProps> = ({
   );
 };
 
-const Wrapper = styled("div")<{ size: "normal" | "small" }>(({ size, theme }) => ({
-  border: `1px solid ${theme.outline.weak}`,
-  borderRadius: "4px", // TODO: use theme value
-  background: theme.bg[1],
-  transition: "all 0.3s", // TODO: use theme value
-  display: "flex",
-  gap: theme.spacing.smallest,
-  alignItems: "center",
-  padding:
-    size === "small"
-      ? `${theme.spacing.micro}px ${theme.spacing.smallest}px`
-      : `${theme.spacing.smallest}px ${theme.spacing.small}px`,
-}));
+const Wrapper = styled("div")<{
+  size: "normal" | "small";
+  appearance?: "readonly" | "present";
+  status: "default" | "active";
+}>(({ size, theme, appearance, status }) => {
+  const borderStyle =
+    appearance === "present"
+      ? status === "default"
+        ? "none"
+        : `1px solid ${theme.select.main}`
+      : status === "active"
+      ? `1px solid ${theme.select.main}`
+      : `1px solid ${theme.outline.weak}`;
 
-const StyledInput = styled("input")(({ theme, disabled }) => ({
-  outline: "none",
-  border: "none",
-  background: "none",
-  color: theme.content.main,
-  // ...theme.typography.body, TODO: update theme
-  // padding: "4px 8px", // TODO: use theme value
-  flex: 1,
-  opacity: disabled ? 0.6 : 1, // TODO: use theme value
-  pointerEvents: disabled ? "none" : "inherit",
-  colorScheme: "dark", // TODO: use theme value
-  // TODO: implement active style on wrapper
-  // ":focus": {
-  //   borderColor: theme.outline.main,
-  // },
-}));
+  return {
+    border: borderStyle,
+    borderRadius: theme.radius.small,
+    background: theme.bg[1],
+    transition: "all 0.3s",
+    display: "flex",
+    gap: theme.spacing.smallest,
+    alignItems: "center",
+    cursor: appearance === "readonly" ? "not-allowed" : "auto",
+    padding:
+      size === "small"
+        ? `${theme.spacing.micro}px ${theme.spacing.smallest}px`
+        : `${theme.spacing.smallest}px ${theme.spacing.small}px`,
+    boxShadow: theme.shadow.input,
+  };
+});
+
+const StyledInput = styled("input")<{ disabled?: boolean; readOnly?: boolean }>(
+  ({ theme, disabled, readOnly }) => ({
+    outline: "none",
+    border: "none",
+    background: "none",
+    color: disabled ? theme.content.weaker : theme.content.main,
+    flex: 1,
+    cursor: disabled ? "not-allowed" : "auto",
+    colorScheme: theme.colorSchema,
+    ...(readOnly && { pointerEvents: "none", userSelect: "none" }),
```

Comment:
We don't need to set pointerEvents and userSelect here. They're still useful even readonly.